test/integration/job/job_test.go

```diff
@@ -1667,6 +1667,165 @@ func TestIndexedJob(t *testing.T) {
 	validateTerminatedPodsTrackingFinalizerMetric(t, 5)
 }
 
+func TestJobPodReplacementPolicy(t *testing.T) {
+	const initialCompletions int32 = 2
+	var podReplacementPolicy = func(obj batchv1.PodReplacementPolicy) *batchv1.PodReplacementPolicy {
+		return &obj
+	}
+	var completionPolicy = func(obj batchv1.CompletionMode) *batchv1.CompletionMode {
+		return &obj
+	}
+	jobSpecIndexedDefault := &batchv1.JobSpec{
+		Parallelism:    pointer.Int32Ptr(initialCompletions),
+		Completions:    pointer.Int32Ptr(initialCompletions),
+		CompletionMode: completionPolicy(batchv1.IndexedCompletion),
+	}
+	cases := map[string]struct {
+		podReplacementPolicyEnabled bool
+		wantTerminating             *int32
+		wantFailed                  int
+		wantActive                  int
+		jobSpec                     *batchv1.JobSpec
+	}{
+		"feature flag off, delete pods and verify no terminating status": {
+			jobSpec:    jobSpecIndexedDefault,
+			wantActive: 2,
+			wantFailed: 2,
+		},
+		"feature flag true, delete pods and verify terminating status": {
+			podReplacementPolicyEnabled: true,
+			wantTerminating:             pointer.Int32(2),
+			wantFailed:                  2,
+			jobSpec:                     jobSpecIndexedDefault,
+		},
+		"feature flag true, delete pods, verify terminating status and recreate upon terminating": {
+			podReplacementPolicyEnabled: true,
+			wantTerminating:             pointer.Int32(2),
+			wantFailed:                  2,
```

Comment:
nit: I think a bit weird that in spec we use `initialCompletions`, and in status we check `2` while the two values are equal. I would suggest to either have one generic constant, such as `podsCount`, or just use numbers. I'm leaning towards the second option slightly.

Reply:
I think I need a constant because completions takes a pointer but I'll leave a general constant such as `podsCount` and use that instead.  Consistency. ;)